packages/mongodb-rag-ingest/src/Config.ts

```diff
@@ -45,6 +45,46 @@ export type Config = {
     Options for the chunker.
    */
   chunkOptions?: Constructor<Partial<ChunkOptions>>;
+
+  /**
+    Options for concurrency.
+   */
+  concurrencyOptions?: Constructor<ConcurrencyOptions>;
 };
 
+
+  /**
+    Options for concurrency. 
+    Set the number of concurrent promises to execute for the given tasks.
+    If not specified, tasks will be run sequentially.
+   */
+export interface ConcurrencyOptions {
+  /**
+    Options for concurrency when chunking and embedding content
+    with the `embed` command.
+   */
+  embed?: {
+    /**
+      Number of pages to chunk and embed concurrently.
+     */
+    processPages?: number,
+    /**
+      Maximum number of chunks per page to generate chunks for concurrently.
+      This includes the creation of the chunk embeddings and any chunk preprocessing.
+     */
+    createChunks?: number,
+  },
+  /**
+    Options for concurrency when ingesting
+    with the `pages` command.
+   */
+  pages?: {
+    /**
+      Number of data sources to process concurrently.
+     */
+    processDataSources?: number,
+  }
+}
```

Comment:
pls add line breaks before all doc comments like below. 

chris had a preference for it. and it's a convention we try to uphold

```suggestion
export interface ConcurrencyOptions {

  /**
    Options for concurrency when chunking and embedding content
    with the `embed` command.
   */
  embed?: {
 
    /**
      Number of pages to chunk and embed concurrently.
     */
    processPages?: number,

    /**
      Maximum number of chunks per page to generate chunks for concurrently.
      This includes the creation of the chunk embeddings and any chunk preprocessing.
     */
    createChunks?: number,
  },

  /**
    Options for concurrency when ingesting
    with the `pages` command.
   */
  pages?: {

    /**
      Number of data sources to process concurrently.
     */
    processDataSources?: number,
  }
}
```